src/interaction/TrackManager.js

```diff
@@ -266,29 +266,28 @@ class TrackManager {
           // control point
           const {deleted, pointBefore, pointAfter, newSegment} = this.trackData_.deleteControlPoint(selected);
 
-        // remove deleted features from source
-        deleted.forEach(f => this.source_.removeFeature(f));
+          // remove deleted features from source
+          deleted.forEach(f => this.source_.removeFeature(f));
 
-        // add newly created segment to source
-        if (newSegment) {
-          this.source_.addFeature(newSegment);
-        }
-
-        // update adjacent points
-        if (pointBefore || pointAfter) {
-          const geometryUpdates = [];
-          if (pointBefore) {
-            geometryUpdates.push(this.updater_.updateAdjacentSegmentsGeometries(pointBefore, this.snapping));
+          // add newly created segment to source
+          if (newSegment) {
+            this.source_.addFeature(newSegment);
           }
-          if (pointAfter) {
-            geometryUpdates.push(this.updater_.updateAdjacentSegmentsGeometries(pointAfter, this.snapping));
+
+          // update adjacent points
+          if (pointBefore || pointAfter) {
+            if (pointBefore) {
+              await this.updater_.updateAdjacentSegmentsGeometries(pointBefore, this.snapping);
+            }
+            if (pointAfter) {
+              await this.updater_.updateAdjacentSegmentsGeometries(pointAfter, this.snapping);
+            }
+            this.onTrackChanged_()
           }
-          Promise.all(geometryUpdates).then(() => this.onTrackChanged_());
```

Comment:
With Promise.all the requests are done in parallel, wich is not the case when using await.
As a consequence I would keep using the old mechanism.

Reply:
done, Promise.all is now also used in `updateAdjacentSegmentsGeometries`